soh/src/overlays/actors/ovl_En_Horse/z_en_horse.c

```diff
@@ -677,34 +678,36 @@ s32 EnHorse_Spawn(EnHorse* this, PlayState* play) {
     Player* player;
     Vec3f spawnPos;
 
-    for (i = 0; i < 169; i++) {
-        if (sHorseSpawns[i].scene == play->sceneNum) {
-            player = GET_PLAYER(play);
-            if (play->sceneNum != SCENE_LON_LON_RANCH ||
-                //! Same flag checked twice
-                (Flags_GetEventChkInf(EVENTCHKINF_EPONA_OBTAINED) && ((gSaveContext.eventInf[0] & 0xF) != 6 || Flags_GetEventChkInf(EVENTCHKINF_EPONA_OBTAINED))) ||
-                // always load two spawns inside lon lon
-                ((sHorseSpawns[i].pos.x == 856 && sHorseSpawns[i].pos.y == 0 && sHorseSpawns[i].pos.z == -918) ||
-                 (sHorseSpawns[i].pos.x == -1003 && sHorseSpawns[i].pos.y == 0 && sHorseSpawns[i].pos.z == -755))) {
-
-                spawnPos.x = sHorseSpawns[i].pos.x;
-                spawnPos.y = sHorseSpawns[i].pos.y;
-                spawnPos.z = sHorseSpawns[i].pos.z;
-                dist = Math3D_Vec3f_DistXYZ(&player->actor.world.pos, &spawnPos);
-
-                if (play->sceneNum) {}
-                if (!(minDist < dist) && !func_80A5BBBC(play, this, &spawnPos)) {
-                    minDist = dist;
-                    this->actor.world.pos.x = sHorseSpawns[i].pos.x;
-                    this->actor.world.pos.y = sHorseSpawns[i].pos.y;
-                    this->actor.world.pos.z = sHorseSpawns[i].pos.z;
-                    this->actor.prevPos = this->actor.world.pos;
-                    this->actor.world.rot.y = sHorseSpawns[i].angle;
-                    this->actor.shape.rot.y = Actor_WorldYawTowardActor(&this->actor, &GET_PLAYER(play)->actor);
-                    spawn = true;
-                    SkinMatrix_Vec3fMtxFMultXYZW(&play->viewProjectionMtxF, &this->actor.world.pos,
-                                                 &this->actor.projectedPos, &this->actor.projectedW);
-                }
+    SceneDBEntry* entry = SceneDB_Retrieve(play->sceneNum);
```

Comment:
Instead of a giant array of valid epona spawn locations, they are stored per-scene.